library/pod-security-policy/users/template.yaml

```diff
@@ -15,55 +15,87 @@ spec:
           properties:
             runAsUser:
               type: object
+              description: "Controls which user ID pod containers can be run with."
               properties:
                 rule:
                   type: string
+                  description: "A strategy for applying the runAsUser restriction."
+                  enum:
+                    - MustRunAs
+                    - MustRunAsNonRoot
+                    - RunAsAny
                 ranges:
                   type: array
+                  description: "A list of ranges affected by the rule."
                   items:
                     type: object
+                    description: "The range of user IDs affected by the rule."
                     properties:
                       min:
                         type: integer
                       max:
                         type: integer
             runAsGroup:
               type: object
+              description: "Controls which group ID pod containers can be run with."
               properties:
                 rule:
                   type: string
+                  description: "A strategy for applying the runAsGroup restriction."
+                  enum:
+                    - MustRunAs
+                    - MayRunAs
+                    - RunAsAny
                 ranges:
                   type: array
+                  description: "A list of ranges affected by the rule."
                   items:
                     type: object
+                    description: "The range of group IDs affected by the rule."
                     properties:
                       min:
                         type: integer
                       max:
                         type: integer
             supplementalGroups:
               type: object
+              description: "Controls which groups are applied to the first process run in any pod container, in addition to the container's primary GID."
```

Comment:
Is there a reason we're calling out the "first process"?

Reply:
I guess "main process" would be a better way to describe this.

I stole this from the [kubernetes `core` code](https://github.com/kubernetes/api/blob/87c4113e35a1b376e4434139109105243860ea52/core/v1/types.go#L3329).

My understanding is that other processes are forked off of that first process, so they can only have groups that it would have?  Is that right?

Reply:
Please reword anything inspired from others' work.

Reply:
Also, I do like "main process" better. You can do escalation via methods like `su`, so it's not strictly true that it's an inheritance tree, but I do think that's the default behavior.

Reply:
New version:

```
              description: "Controls which groups are applied to a pod container's main process, in addition to the container's primary GID."
```

Reply:
After reading other comments, I think it's better to go a [simpler route](https://github.com/open-policy-agent/gatekeeper-library/pull/109#discussion_r714288228), where we don't try to teach the feature in the CT.  Rather, we should say specifically what policy the CT enforces and no more.